commit 4fa32f25342398e9d5f257313f7442c4e016b3b9 (refs/changes/09/2409/4)
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2017-08-15T16:54:47-07:00 (2 years, 2 months ago)
    
    CNAPI-714 zones in nightly1 with cpu_caps set to 'null' causing test-servers test to fail
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/test/test-servers.js b/test/test-servers.js
index 05df839..2dd560e 100644
--- a/test/test-servers.js
+++ b/test/test-servers.js
@@ -362,13 +362,20 @@ function validateServer(t, server, options) {
 
             t.ok(UUID_RE.test(vm.owner_uuid), 'ensure vm.owner_uuid is a UUID');
 
-            var numAttr = ['max_physical_memory', 'quota', 'cpu_cap'];
+            var numAttr = ['max_physical_memory', 'quota'];
             numAttr.forEach(function (attr) {
                 if (typeof (vm[attr]) !== 'undefined') {
                     vmAttrTypeEqual(t, vm, attr, 'number');
                 }
             });
 
+            var optionalNumAttr = ['cpu_cap'];
+            optionalNumAttr.forEach(function (attr) {
+                if (vm[attr] !== undefined && vm[attr] !== null) {
+                    vmAttrTypeEqual(t, vm, attr, 'number');
+                }
+            });
+
             vmAttrTypeEqual(t, vm, 'last_modified', 'string');
             vmAttrTypeEqual(t, vm, 'state', 'string');
         }
